commons/src/main/com/mbrlabs/mundus/commons/utils/DebugRenderer.java

```diff
@@ -129,6 +137,28 @@ public void render(GameObject go) {
             instances.add(modelInstance);
         }
 
+        if (drawFacingArrow && go == selectedGameObject){
+            //TODO Cache arrow model
+            Vector3 origin = new Vector3();
```

Comment:
You can get these object from [Pool](https://github.com/JamesTKhan/Mundus/blob/master/commons/src/main/com/mbrlabs/mundus/commons/utils/Pools.java) and you don't need to create new object in a render method.

Reply:
Thanks for the Pool() tip! This PR is ready.

Reply:
And don't forget to call Pools.vector3Pool.free(origin) and Pools.vector3Pool.free(facing) end of block (about line 157). This method add back the object to the pool and can be used the object again if call obtain method. If you don't call free method, like now, the GC will delete these object.